From 4183fbbb8f9d8fbc1266a77867b586bfea0c0e5c Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Mon, 6 Feb 2017 12:07:41 -0800
Subject: [PATCH] joyent/node-mname#13 large amount of time spent in modulo
 operations Reviewed by: Cody Peter Mello <cody.mello@joyent.com>

---
 lib/dns-buffer.js | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/dns-buffer.js b/lib/dns-buffer.js
index 5d3a3bc..08ab3ac 100644
--- a/lib/dns-buffer.js
+++ b/lib/dns-buffer.js
@@ -191,21 +191,21 @@ DNSBuffer.prototype.writeNamePlain = function (name) {
 DNSBuffer.prototype.writeUInt32 = function (v) {
         while (this._offset + 4 > this._size)
                 this.expand();
-        this._buffer.writeUInt32BE(v % 0xffffffff, this._offset);
+        this._buffer.writeUInt32BE(v, this._offset);
         this._offset += 4;
 };
 
 DNSBuffer.prototype.writeUInt16 = function (v) {
         while (this._offset + 2 > this._size)
                 this.expand();
-        this._buffer.writeUInt16BE(v & 0xffff, this._offset);
+        this._buffer.writeUInt16BE(v, this._offset);
         this._offset += 2;
 };
 
 DNSBuffer.prototype.writeUInt8 = function (v) {
         while (this._offset + 1 > this._size)
                 this.expand();
-        this._buffer[this._offset++] = v & 0xff;
+        this._buffer.writeUInt8(v, this._offset++);
 };
 
 DNSBuffer.prototype.write = function (buf) {
-- 
2.21.0

